symbolic-debuginfo/src/pe.rs

```diff
@@ -209,14 +212,30 @@ impl<'data> PeObject<'data> {
         false
     }
 
-    /// Constructs a no-op debugging session.
-    pub fn debug_session(&self) -> Result<PeDebugSession<'data>, PeError> {
-        Ok(PeDebugSession { _ph: PhantomData })
+    /// Constructs a debugging session.
+    ///
+    /// A debugging session loads certain information from the object file and creates caches for
+    /// efficient access to various records in the debug information. Since this can be quite a
+    /// costly process, try to reuse the debugging session as long as possible.
+    ///
+    /// PE files usually don't have embedded debugging information,
+    /// but some toolchains (e.g. MinGW) generate DWARF debug info.
+    ///
+    /// Constructing this session will also work if the object does not contain debugging
+    /// information, in which case the session will be a no-op. This can be checked via
+    /// [`has_debug_info`](struct.PeObject.html#method.has_debug_info).
+    pub fn debug_session(&self) -> Result<DwarfDebugSession<'data>, DwarfError> {
+        let symbols = self.symbol_map();
+        DwarfDebugSession::parse(self, symbols, self.load_address() as i64, self.kind())
     }
 
     /// Determines whether this object contains stack unwinding information.
     pub fn has_unwind_info(&self) -> bool {
+        // Sections don't exist in "vanilla" PE, only one w/ DWARF
+        // FIXME: Is this change needed / what uses this functionality?
         !self.is_stub && self.exception_data().map_or(false, |e| !e.is_empty())
+            || self.has_section("eh_frame")
+            || self.has_section("debug_frame")
```

Comment:
Our CFI converter is using this. I believe we can skip checking for `eh_frame` and `debug_frame` as long as we cannot actually consume that data, which I believe we do not at this point.

Reply:
Alright, removed this for now.